services/catarse/app/models/concerns/user/custom_validators.rb

```diff
@@ -4,10 +4,18 @@ module User::CustomValidators
   extend ActiveSupport::Concern
 
   included do
-    validate :no_base64_images
+    validate :no_base64_images, :block_email_with_reference_to_catarse
 
     def no_base64_images
       errors.add(:about_html, :base64_images_not_allowed) if about_html.try(:match?, 'data:image/.*;base64')
     end
+
+    def block_email_with_reference_to_catarse
+      return unless email_changed?
+      if email.include? '@catarse'
+        errors.add(:email, :email_not_allowed)
```

Comment:
**Problema:**
- Aqui é melhor tu verificar na primeira condição que o e-mail está presente pra não tentar chamar os métodos seguintes num `nil`.
- E também fazer um `email.split('@').last.try(:include?, 'catarse')` pra verificar se tem catarse em qualquer parte do domínio e não apenas no início. Assim a gente consegue evitar casos como `admin@suporte.catarse.me` por exemplo.
- Usar o error :invalid do rails pra não precisar criar uma mensagem nova, pois a mensagem padrão já se encaixa bem nesse caso.
```suggestion
    def block_email_with_reference_to_catarse
      return unless email.present? && email_changed?
      if email.split('@').last.try(:include?, 'catarse')
        errors.add(:email, :invalid)
```